api/v1alpha3/types.go

```diff
@@ -118,9 +118,12 @@ const (
 
 // IngressRule defines an Azure ingress rule for security groups.
 type IngressRule struct {
+	Name        string                `json:"name"`
 	Description string                `json:"description"`
 	Protocol    SecurityGroupProtocol `json:"protocol"`
 
+	Priority int32 `json:"priority,omitempty"`
```

Comment:
let's add a comment/description of priority + some validation. According to Azure docs it should be:

```
A number between 100 and 4096. Rules are processed in priority order, with lower numbers processed before higher numbers, because lower numbers have higher priority. Once traffic matches a rule, processing stops. As a result, any rules that exist with lower priorities (higher numbers) that have the same attributes as rules with higher priorities are not processed.
```

Each rule should also have a unique priority.

Reply:
Fixed this bit in an amended commit.